### PR TITLE
User token list

### PR DESCRIPTION
**Pull request by sschneid**
Add a `GET /api/v1/token` route, eg:

````
$ curl -X GET -u jdoe --url https://vmpooler/api/v1/token
Enter host password for user 'jdoe':
````
````json
{
  "ok": "true",
  "a9znth9dq01t416hrguu56ze37t790bl": {
    "created": "2015-04-30 19:09:20 -0700"
  },
  "ikqe0mtk7zzzjljvbgxnocx8w1o9dc9n": {
    "created": "2015-05-19 13:25:42 -0700"
  }
}
````

### CODE REVIEW

**Comment by sschneid** [comment]
This is deployed on vmpooler-dev alongside #125.

File: lib/vmpooler/api/v1.rb [hunk 1]
```diff
@@ -162,6 +162,41 @@ def need_token!
...
+
+      status 404
+      result = { 'ok' => false }
+
+      if Vmpooler::API.settings.config[:auth]
+        status 401
+
+        need_auth!
+
+        backend.keys('vmpooler__token__*').each do |key|
```
Comment (justinstoller) [inline lib/vmpooler/api/v1.rb]: Dev facing docs on the data model in Redis, or an abstraction over it, would be awesome for coming up to speed on pooler development.

IOW, from elsewhere in this file this _looks_ right. But I don't know what I don't know, ya know?

File: lib/vmpooler/api/v1.rb [hunk 2]
```diff
@@ -162,6 +162,41 @@ def need_token!
...
+          data = backend.hgetall(key)
+
+          token = key.split('__').last
```
Comment (justinstoller) [inline lib/vmpooler/api/v1.rb]: this could be moved inside the conditional right?

Comment (sschneid) [inline lib/vmpooler/api/v1.rb]: > this could be moved inside the conditional right?

Updated.

**Comment by justinstoller** [comment]
:+1:

**Pull request merged**